main.go

```diff
@@ -103,25 +103,37 @@ func run(l *log.Logger, args arguments) error {
 		l.Println(string(output))
 	}
 
-	var out io.Writer
-	if args.outputFile == "-" {
-		out = os.Stdout
+	var input io.Reader
+	if args.templateFile == "-" {
+		input = os.Stdin
 	} else {
-		var err error
-		out, err = os.Create(args.outputFile)
+		f, err := os.Open(args.templateFile)
 		if err != nil {
 			return err
 		}
+
+		defer f.Close()
```

Comment:
I don't report the error here - standard library doesn't either. It's pretty meaningless, I think, if we've managed to _read_ as much as we want successfully?

Reply:
Yes, I think this is fine.